src/core/api/api.py

```diff
@@ -584,3 +589,80 @@ def get_objects_from_job(
     # convert to schema Object's:
     data["data"] = [schema.Object(**o.to_api()) for o in data["data"]]
     return data
+
+
+@core_api.get("/storage", response_model=List[Union[Dict[str, Any], str, None]])
+@core_api.get(
+    "/storage/", response_model=List[Union[Dict[str, Any], str, None]]
+)
+async def get_storage() -> List[Union[Dict[str, Any], str, None]]:
+    """Get directory listing for a given path to a directory in jsTree json format.
+
+    Parameters
+    ----------
+    path:   str
+        Path to a folder as a string.
+
+    Returns
+    -------
+    Dict[str, Any]
+        Json data explaining the file structure in jsTree format for a folder.
+        Does not run recursivly through all subfolders.
+
+    Raises
+    ------
+    HTTPException
+        If the path is a file, and not a directory.
+    HTTPException
+        If the path is invalid.
+    """
+    path = pathlib.Path(sys.executable).anchor
```

Comment:
Make an attempt at setting this using the new config feature of #73.

We've looked at using the startup event from FastAPI/Starlette.  But have been unable to override a global root_path.

Ref.: https://fastapi.tiangolo.com/advanced/testing-events/

```diff
diff --git a/src/core/api/api.py b/src/core/api/api.py
index bf1b5db..daed219 100644
--- a/src/core/api/api.py
+++ b/src/core/api/api.py
@@ -10,7 +10,7 @@ import base64
 import logging
 import pathlib
 import sys
-from typing import Any, AsyncGenerator, Dict, Generator, List, Union
+from typing import Any, AsyncGenerator, Dict, Generator, List, Optional, Union
 
 from fastapi import (
     Depends,
@@ -39,6 +39,16 @@ logger = logging.getLogger(__name__)
 
 core_api = FastAPI()
 
+root_path: Optional[str] = None
+# root_path: Optional[str] = pathlib.Path(sys.executable).anchor
+
+
+@core_api.on_event("startup")
+async def startup_event():
+    global root_path
+    root_path = pathlib.Path(sys.executable).anchor
+    print(f"Event in api.py")
+
 
 def get_runtime_repo() -> Generator[ProjectRepository, None, None]:
     """Fastapi dependencies function creating `repositories` for endpoint."""
@@ -616,17 +626,15 @@ async def get_storage() -> List[Union[Dict[str, Any], str, None]]:
     HTTPException
         If the path is invalid.
     """
-    path = pathlib.Path(sys.executable).anchor
-
     try:
-        return get_directory_listing(path)
+        return get_directory_listing(root_path)
     except NotADirectoryError:
-        logger.warning("Chosen path '{}' is not a directory.".format(path))
+        logger.warning("Chosen path '{}' is not a directory.".format(root_path))
         raise HTTPException(
             status_code=400, detail="Chosen path is not a directory"
         )
     except FileNotFoundError:
-        logger.warning("Chosen path '{}' is not valid.".format(path))
+        logger.warning("Chosen path '{}' is not valid.".format(root_path))
         raise HTTPException(status_code=404, detail="Chosen path is not valid")
 
 
diff --git a/src/core/services.py b/src/core/services.py
index b33657a..d212aa8 100644
--- a/src/core/services.py
+++ b/src/core/services.py
@@ -591,11 +591,11 @@ def get_directory_listing(
     Dict
         jsTree formatted json containing information about root node at path.
     """
-    if path is None:
-        # https://stackoverflow.com/a/65975115/182868
-        # TODO: The root folders should have been set via the new config
-        # system.
-        path = pathlib.Path(sys.executable).anchor
+    # if path is None:
+    #     # https://stackoverflow.com/a/65975115/182868
+    #     # TODO: The root folders should have been set via the new config
+    #     # system.
+    #     path = pathlib.Path(sys.executable).anchor
 
     normalized_path = pathlib.Path(path)
 
diff --git a/tests/unit/core/integration/test_api.py b/tests/unit/core/integration/test_api.py
index dee2781..b0119aa 100644
--- a/tests/unit/core/integration/test_api.py
+++ b/tests/unit/core/integration/test_api.py
@@ -10,6 +10,7 @@ from sqlalchemy.orm import clear_mappers
 from sqlalchemy.orm.session import close_all_sessions
 
 import core
+import core.api
 import core.main
 from core import api, model
 from core.repository import SqlAlchemyProjectRepository as ProjectRepository
@@ -27,6 +28,19 @@ TEST_VIDEO = str(
 )
 
 
+async def test_startup_event():
+    global root_path
+    root_path = "tests/integration/test_data/test_directory_listing_folder"
+    api.root_path = "tests/integration/test_data/test_directory_listing_folder"
+    core.api.root_path = (
+        "tests/integration/test_data/test_directory_listing_folder"
+    )
+    print(f"Event in test_apy.py {api.root_path}")
+
+
+api.core_api.router.on_startup = [test_startup_event]
+
+
 @pytest.fixture
 def setup(tmp_path):
     """Override setup of db for tests."""
@@ -603,6 +617,53 @@ def test_get_job_objects_no_job_project(make_test_data) -> None:
 
 
 def test_get_storage():
+    """Test getting the directory listing from the API."""
+    with TestClient(api.core_api) as client:
+        response = client.get("/storage/")
+        response_data = response.json()
+        print(response_data)
+        assert response.status_code == 200
+        assert response_data[0]["text"] == "test_directory_listing_folder"
+        assert response_data[0]["type"] == "folder"
+        assert "children" in response_data[0]
+
+        children_results = [
+            {
+                "id": "tests/integration/test_data/test_directory_listing_folder/folderA",
+                "text": "folderA",
+                "type": "folder",
+                "children": True,
+            },
+            {
+                "id": "tests/integration/test_data/test_directory_listing_folder/folderB",
+                "text": "folderB",
+                "type": "folder",
+                "children": True,
+            },
+            {
+                "id": "tests/integration/test_data/test_directory_listing_folder/emptyfile",
+                "text": "emptyfile",
+                "type": "file",
+            },
+            {
+                "id": "tests/integration/test_data/test_directory_listing_folder/invalidext.in21p3",
+                "text": "invalidext.in21p3",
+                "type": "file",
+            },
+            {
+                "id": "tests/integration/test_data/test_directory_listing_folder/text.txt",
+                "text": "text.txt",
+                "type": "text/plain",
+            },
+            {
+                "id": "tests/integration/test_data/test_directory_listing_folder/video.mp4",
+                "text": "video.mp4",
+                "type": "video/mp4",
+            },
+        ]
+
+
+def test_get_storage_path():
     """Test getting the directory listing from the API."""
     with TestClient(api.core_api) as client:
         pathb = (
```